app/gen-server/lib/Housekeeper.ts

```diff
@@ -59,13 +63,22 @@ export class Housekeeper {
     this._logMetricsInterval = setInterval(() => {
       this.logMetricsExclusively().catch(log.warn.bind(log));
     }, LOG_METRICS_PERIOD_MS);
+
+    await this.checkVersionUpdatesExclusively();
```

Comment:
I think this is wrong.  Now, everytime we start the server, it will try to call some API, and we pay the cost at this right moment.

It looks like you needed it just for test, to somehow ensure that the interval is triggered.

For test like that, we can either expose a test method (that will just perform the check) or change the interval, for every second, and just wait 2 seconds for the call.

Reply:
No, it's intentional. It's not for testing. I do want this at startup. With the API delay.

There will be config option available to disable version checking, but it will be on by default.

Reply:
Wdyt about inverting the logic, checking it only when the variable is set? I'm worried about electron/tests/grist-static code that now will do request by default.

Reply:
@Spoffy Sure, let's run it aync without blocking. Done.

@berhalak I'll soon prepare a kill switch so the automated checking can be turned off, and I'll add that switch to tests and desktop and static.